internal/rest/leveldbreceipts.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2021 Kaleido
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rest
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/kaleido-io/ethconnect/internal/errors"
+	"github.com/kaleido-io/ethconnect/internal/kvstore"
+)
+
+type levelDBReceipts struct {
+	conf  *LevelDBReceiptStoreConf
+	store kvstore.KVStore
+}
+
+func newLevelDBReceipts(conf *LevelDBReceiptStoreConf) (*levelDBReceipts, error) {
+	store, err := kvstore.NewLDBKeyValueStore(conf.Path)
+	if err != nil {
+		return nil, errors.Errorf(errors.ReceiptStoreLevelDBConnect, err)
+	}
+	return &levelDBReceipts{
+		conf:  conf,
+		store: store,
+	}, nil
+}
+
+// AddReceipt processes an individual reply message, and contains all errors
+// To account for any transitory failures writing to mongoDB, it retries adding receipt with a backoff
+func (l *levelDBReceipts) AddReceipt(requestID string, receipt *map[string]interface{}) (err error) {
+	b, _ := json.MarshalIndent(&receipt, "", "  ")
+	return l.store.Put(requestID, b)
+}
+
+// GetReceipts Returns recent receipts with skip & limit
+func (l *levelDBReceipts) GetReceipts(skip, limit int, ids []string, sinceEpochMS int64, from, to string) (*[]map[string]interface{}, error) {
+	itr := l.store.NewIterator()
```

Comment:
@jimthematrix - sorry if I've missed it, but what ensures that this is in reverse addition order?
I'm not sure that the natural order of the `requestID` strings would do that automatically.

Reply:
good question, I was thinking about that as well. 

Given that the iteration order is by default based on key lexicon order, we need to set a different order using a comparator. I'm thinking to use `receivedAt` as the first level check, but since its precision is millisecond, we need a tie breaker of the `timeElapsed` which is nanosecond. This way we don't have to implement a synthetic index

Reply:
guess the Comparer only operates on keys, not values. so the above won't work. may have to think about using a composite key assembled from the two fields mentioned above